zaza/charm_tests/test_utils.py

```diff
@@ -71,23 +71,43 @@ def config_change(self, default_config, alternate_config):
         :param alternate_config: Dict of charm settings to change to
         :type alternate_config: dict
         """
-        logging.debug('Changing charm setting to {}'.format(alternate_config))
-        model.set_application_config(
-            self.application_name,
-            alternate_config,
-            model_name=self.model_name)
+        # we need to compare config values to what is already applied before
+        # attempting to set them.  otherwise the model will behave differently
+        # than we would expect while waiting for completion of the change
+        _app_config = model.get_application_config(self.application_name)
+        app_config = {}
+        for k, v in _app_config.items():
+            app_config[k] = str(v.get('value', ''))
+        if all(item in app_config.items()
+                for item in alternate_config.items()):
+            logging.debug('alternate_config equals what is already applied '
+                          'config')
+            yield
+            if default_config == alternate_config:
```

Comment:
Are there any cases where we would have a nested dictionary? I guess not.

Reply:
Good question, I think the cases where we have those kinds of configuration options the user would have to provide it as a YAML- or JSON-encoded `str` and not a actual dict, so we should be home free.